### ISSUE TITLE
java.lang.IllegalArgumentException: @$Proxy6: one of [uiAutomator, accessibility, className, id, tagName, name, xpath, linkText, partialLinkText, windowsAutomation, iOSClassChain, iOSNsPredicate] should be filled

### ISSUE THREAD

**Issue by avanishf** [appium/java-client#725]
## Description
I am stuck with this issue while running a sample TestNG test case, it throws an IllegalArgumentException

## Environment
* java client build version : 5.0.1
* Appium server version : 1.6.5
* Desktop OS/version used to run Appium if necessary: MacOSSierra
* Mobile platform/version under test: iOS 10.3
* Real device or emulator/simulator: iPhone 7 Simulator

## Details

The same code works for Android but not for iOS. This issue i face is only for SignupScreen class. Somehow its not  able to initialise elements for SignupScreen. But for other custom classes, iOS test runs fine.

Here is my sample TestNG test which i am trying to run : 
https://gist.github.com/avanishf/6a460188cbcb4ed81b42969f7faf76a4

And here is my SignupScreen class : 
https://gist.github.com/avanishf/d4eb398ded0865649a82b99139dcfca4

## Ecxeption stacktraces
``
java.lang.IllegalArgumentException: @$Proxy6: one of [uiAutomator, accessibility, className, id, tagName, name, xpath, linkText, partialLinkText, windowsAutomation, iOSClassChain, iOSNsPredicate] should be filled
	at io.appium.java_client.pagefactory.bys.builder.AppiumByBuilder.getFilledValue(AppiumByBuilder.java:109)
	at io.appium.java_client.pagefactory.bys.builder.AppiumByBuilder.createBy(AppiumByBuilder.java:148)
	at io.appium.java_client.pagefactory.DefaultElementByBuilder.getBys(DefaultElementByBuilder.java:133)
	at io.appium.java_client.pagefactory.DefaultElementByBuilder.buildMobileNativeBy(DefaultElementByBuilder.java:187)
	at io.appium.java_client.pagefactory.DefaultElementByBuilder.buildBy(DefaultElementByBuilder.java:216)
	at io.appium.java_client.pagefactory.AppiumElementLocatorFactory.createLocator(AppiumElementLocatorFactory.java:66)
	at io.appium.java_client.pagefactory.AppiumElementLocatorFactory.createLocator(AppiumElementLocatorFactory.java:53)
	at io.appium.java_client.pagefactory.AppiumElementLocatorFactory.createLocator(AppiumElementLocatorFactory.java:1)
	at org.openqa.selenium.support.pagefactory.DefaultFieldDecorator.decorate(DefaultFieldDecorator.java:56)
	at io.appium.java_client.pagefactory.AppiumFieldDecorator.decorate(AppiumFieldDecorator.java:157)
	at org.openqa.selenium.support.PageFactory.proxyFields(PageFactory.java:113)
	at org.openqa.selenium.support.PageFactory.initElements(PageFactory.java:105)
	at pageObjects.SignupScreen.<init>(SignupScreen.java:18)
	at testCases.sampleTestNGTest.sampleTest(sampleTestNGTest.java:26)
``

## Link to Appium logs

``
60,"shouldUseSingletonTestManager":true}}
[JSONWP Proxy] Got response with status 200: {"value":{"sessionId":"D673AC1D-0597-4591-9B07-012BF4C51DE7","capabilities":{"device":"iphone","browserName":"Sniip2-Testing","sdkVersion":"10.3.1","CFBundleIdentifier":"com.digitalassetgroup.sniip-testing"}},"sessionId":"D673AC1D-0597-4591-9B07-012BF4C51DE7","status":0}
[BaseDriver] Event 'wdaSessionStarted' logged at 1505359344889 (13:22:24 GMT+1000 (AEST))
[XCUITest] Found WDA derived data folder: '/Users/avanishf/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs'
[XCUITest] Setting '555' permissions to '/Users/avanishf/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs/Logs/Test/Attachments' folder
[BaseDriver] Event 'wdaPermsAdjusted' logged at 1505359344892 (13:22:24 GMT+1000 (AEST))
[BaseDriver] Event 'wdaStarted' logged at 1505359344893 (13:22:24 GMT+1000 (AEST))
[XCUITest] Setting initial orientation to 'PORTRAIT'
[JSONWP Proxy] Proxying [POST /orientation] to [POST http://localhost:8100/session/D673AC1D-0597-4591-9B07-012BF4C51DE7/orientation] with body: {"orientation":"PORTRAIT"}
[JSONWP Proxy] Got response with status 200: {"value":{},"sessionId":"D673AC1D-0597-4591-9B07-012BF4C51DE7","status":0}
[BaseDriver] Event 'orientationSet' logged at 1505359345137 (13:22:25 GMT+1000 (AEST))
[Appium] New XCUITestDriver session created successfully, session a88ceeda-5a44-4b1a-a3ae-4929e98bbeba added to master session list
[BaseDriver] Event 'newSessionStarted' logged at 1505359345140 (13:22:25 GMT+1000 (AEST))
[MJSONWP] Responding to client with driver.createSession() result: {"webStorageEnabled":false,"locationContextEnabled":false,"browserName":"","platform":"MAC","javascriptEnabled":true,"databaseEnabled":false,"takesScreenshot":true,"networkConnectionEnabled":false,"app":"/Users/avanishf/Desktop/iOS app/Sniip2-Testing.app","clearSystemFiles":true,"platformVersion":"10.3","platformName":"iOS","deviceName":"iPhone 7","udid":"287EEB9F-BB5A-4B5C-BD62-273144E019BA"}
[HTTP] <-- POST /wd/hub/session 200 25168 ms - 468 
[HTTP] --> GET /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba {}
[MJSONWP] Calling AppiumDriver.getSession() with args: ["a88ceeda-5a44-4b1a-a3ae-4929e98bbeba"]
[XCUITest] Executing command 'getSession'
[JSONWP Proxy] Proxying [GET /] to [GET http://localhost:8100/session/D673AC1D-0597-4591-9B07-012BF4C51DE7] with no body
[JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"sessionId\" : \"D673AC1D-0597-4591-9B07-012BF4C51DE7\",\n    \"capabilities\" : {\n      \"device\" : \"iphone\",\n      \"browserName\" : \"Sniip2-Testing\",\n      \"sdkVersion\" : \"10.3.1\",\n      \"CFBundleIdentifier\" : \"com.digitalassetgroup.sniip-testing\"\n    }\n  },\n  \"sessionId\" : \"D673AC1D-0597-4591-9B07-012BF4C51DE7\",\n  \"status\" : 0\n}"
[XCUITest] Merging WDA caps over Appium caps for session detail response
[MJSONWP] Responding to client with driver.getSession() result: {"udid":"287EEB9F-BB5A-4B5C-BD62-273144E019BA","app":"/Users/avanishf/Desktop/iOS app/Sniip2-Testing.app","clearSystemFiles":true,"platformVersion":"10.3","platformName":"iOS","deviceName":"iPhone 7","device":"iphone","browserName":"Sniip2-Testing","sdkVersion":"10.3.1","CFBundleIdentifier":"com.digitalassetgroup.sniip-testing"}
[HTTP] <-- GET /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba 200 122 ms - 402 
[HTTP] --> GET /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba {}
[MJSONWP] Calling AppiumDriver.getSession() with args: ["a88ceeda-5a44-4b1a-a3ae-4929e98bbeba"]
[XCUITest] Executing command 'getSession'
[JSONWP Proxy] Proxying [GET /] to [GET http://localhost:8100/session/D673AC1D-0597-4591-9B07-012BF4C51DE7] with no body
[JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"sessionId\" : \"D673AC1D-0597-4591-9B07-012BF4C51DE7\",\n    \"capabilities\" : {\n      \"device\" : \"iphone\",\n      \"browserName\" : \"Sniip2-Testing\",\n      \"sdkVersion\" : \"10.3.1\",\n      \"CFBundleIdentifier\" : \"com.digitalassetgroup.sniip-testing\"\n    }\n  },\n  \"sessionId\" : \"D673AC1D-0597-4591-9B07-012BF4C51DE7\",\n  \"status\" : 0\n}"
[XCUITest] Merging WDA caps over Appium caps for session detail response
[MJSONWP] Responding to client with driver.getSession() result: {"udid":"287EEB9F-BB5A-4B5C-BD62-273144E019BA","app":"/Users/avanishf/Desktop/iOS app/Sniip2-Testing.app","clearSystemFiles":true,"platformVersion":"10.3","platformName":"iOS","deviceName":"iPhone 7","device":"iphone","browserName":"Sniip2-Testing","sdkVersion":"10.3.1","CFBundleIdentifier":"com.digitalassetgroup.sniip-testing"}
[HTTP] <-- GET /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba 200 100 ms - 402 
[HTTP] --> DELETE /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba {}
[MJSONWP] Calling AppiumDriver.deleteSession() with args: ["a88ceeda-5a44-4b1a-a3ae-4929e98bbeba"]
[BaseDriver] Event 'quitSessionRequested' logged at 1505359346157 (13:22:26 GMT+1000 (AEST))
[JSONWP Proxy] Proxying [DELETE /session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba] to [DELETE http://localhost:8100/session/D673AC1D-0597-4591-9B07-012BF4C51DE7] with no body
[JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n\n  },\n  \"sessionId\" : \"1DF1C1DA-BF93-4990-AF9E-2B9FD662B266\",\n  \"status\" : 0\n}"
[XCUITest] Shutting down sub-processes
[XCUITest] Shutting down xcodebuild process (pid 52350)
[XCUITest] xcodebuild exited with code 'null' and signal 'SIGTERM'
[XCUITest] Found WDA derived data folder: '/Users/avanishf/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs'
[XCUITest] Setting '755' permissions to '/Users/avanishf/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs/Logs/Test/Attachments' folder
[iOS] Clearing log files
[iOS] Deleting '/Users/avanishf/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs/Logs'. Freeing 208K.
[iOS] Finished clearing log files
[iOSLog] Stopping iOS log capture
[Appium] Removing session a88ceeda-5a44-4b1a-a3ae-4929e98bbeba from our master session list
[BaseDriver] Event 'quitSessionFinished' logged at 1505359346640 (13:22:26 GMT+1000 (AEST))
[MJSONWP] Received response: null
[MJSONWP] But deleting session, so not returning
[MJSONWP] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/a88ceeda-5a44-4b1a-a3ae-4929e98bbeba 200 485 ms - 76
``

**Comment by TikhomirovSergey** [comment]
@avanishf 
This is not the issue of appium. The cause is here
https://gist.github.com/avanishf/d4eb398ded0865649a82b99139dcfca4#file-signupscreen-java-L46

Where should be not an empty string

**Comment by avanishf** [comment]
Thanks Sergey!
